documentation/puppet-api/v4/compile.markdown

```diff
@@ -0,0 +1,93 @@
+---
+layout: default
+title: "Puppet Server: Puppet API: Compile"
+canonical: "/puppetserver/latest/puppet-api/v4/compile.html"
+---
+
+The compile endpoint allows compilation of arbitrary pcore serialized Puppet
+Code ASTs (aka parsed, but unevaluated Puppet Code snippets). In addition to
+the code AST, the caller must provide the certname, environment, facts,
+trusted facts, and variables for compilation (the context in which this AST
+will be evaluated). Optionally, the caller may provide job_id, transaction_id,
+and configure returned logging. The request body must be JSON formatted and
+the caller must accept a JSON response. The server must not have rich_data
+disabled.
```

Comment:
Assuming it's phrased like this because it's enabled by default?

Reply:
yep, should I change the way it's worded?

Reply:
No I think that makes sense given the default. And we call out that default elsewhere on this page.